### PR TITLE
Updated docs for multilingual sites to include a custom middleware

### PR DESCRIPTION
**Pull request by Proper-Job**
Hi there

I noticed while building a multilingual site that the documentation should include mention of activating the language for each page tree.  This is a stab at filling in the missing pieces.

Greetings
Moritz

### CODE REVIEW

**Comment by gasman** [comment]
Thanks @Proper-Job! All looks good to me, but as I haven't worked on a multi-language site, passing on to @kaedroho to confirm that this is the correct approach.

File: docs/advanced_topics/i18n/duplicate_tree.rst [hunk 1]
```diff
@@ -46,6 +54,22 @@ We must set Django's ``LANGUAGES`` setting so we don't redirect non English/Fren
 
             return HttpResponseRedirect(self.url + language + '/')
 
+    # your-app.middleware.py
+    from django.utils import translation
+    from django.utils.translation import get_language_from_request
+
+    class LocaleMiddleware(object):
```
Comment (kaedroho) [inline docs/advanced_topics/i18n/duplicate_tree.rst]: How does this differ from Django's builtin ``LocaleMiddleware``?

https://docs.djangoproject.com/en/1.11/ref/middleware/#module-django.middleware.locale

**Comment by gasman** [comment]
Closing due to lack of updates - if indeed Django's own `LocaleMiddleware` is the preferred solution, a new PR advising developers to use that would be very much welcomed.

**Comment by Proper-Job** [comment]
Sorry guys, I somehow missed all updates on this thread until now.

@kaedroho the difference between this custom middleware and Django's LocaleMiddleware is that in a Wagtail duplicate page tree scenario you most likely don't have [i18n_patterns](https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#language-prefix-in-url-patterns) enabled in your urls.py.  Therefore Django's LocaleMiddleware won't pick the language from the path as you can see [here](https://github.com/django/django/blob/1.11.9/django/middleware/locale.py#L25), because i18n_patterns_used is False.

**Comment by gasman** [comment]
Sorry for the slow progress of this PR... just to note that the code example will now need updating to use the Django >1.11 middleware API.

**Comment by gasman** [comment]
Closing as outdated - these docs were deleted in https://github.com/wagtail/wagtail/commit/16bf4cac56684401734201c9bf2aa7d9ce583ac3 and Wagtail's i18n support has changed dramatically since then.

**Pull request closed without merging**